### ISSUE TITLE
`rm --trash` should be the default behavior

### ISSUE THREAD

**Issue by bobhy** [nushell/nushell#11842]
### Related problem

Re [this discussion](https://github.com/nushell/nushell/pull/11621#issuecomment-1938874208) concerning  globbing arguments of `rm`, it turns out that nushell already has the feature that would add the desired safety without compromising convenience: `rm --trash`.

### Describe the solution you'd like

1. make `--trash` the default on *all* platforms
2. make `--trash` on IOS integrate with the "Recently Deleted" folder, [described here](https://www.anysoftwaretools.com/empty-iphone-trash/)
3. make `--trash` on Android integrate with the "Trash" folder on that platform.  (I don't know when this was added, but it is present on my Pixel running Android 14.)
4. add a nushell command to *empty* the recycle bin on all supported platforms.  Whether this is should be a  brand new command or some switches on existing commands is TBD.

### Describe alternatives you've considered

It's a fair point that having `rm` glob its arguments is a potential foot gun.  But having it *not* glob arguments seems unnecessarily inconvenient if there's a reasonable (and well-known) recovery mechanism (Recycle Bin).

I considered whether nushell should add a config option to control whether `--trash` or `--permanent` should be the default behavior, but decided against proposing it.  Adding the option means this feature would not have to be documented as a breaking change.  However, breaking changes are still acceptable pre V1, and the direction of the change is to improve user safety so it seems reasonable in this case.  The consequence of this breaking change that should be documented would be that storage may unexpected fill up with files you `rm`'ed, and 2) you should be sure to use `rm --permanent` to delete sensitive files.

**Comment by alxpettit** [comment]
I agree with this. It's best to err on the side of caution with our defaults.

**Comment by zyppe** [comment]
I have a question about nushell's 'rm' command. Other shell such as powershell or bash etc, 'rm -rvf' will verbose every file deleted, but nushell will only verbose the folder. This is very dangerous. Is this a nushell's feature or bug that cannot verbose every file deleted?
